js/coinsbit.js

```diff
@@ -0,0 +1,735 @@
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        params = this.handleMarketTypeAndParams ('fetchOHLCV', undefined, params)[1];
+        const request = {
+            'market': market['id']
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default = 50, minimum = 50, maximum = 1000
+        }
+        const defaultMethod = this.safeString (this.options, 'fetchTradesMethod');
+        const method = this.safeString (params, 'method', defaultMethod);
+        let data = {};
+        if (market['spot']) {
+            if (method === 'spotPublicGetHistory') {
+                if (!('lastId' in params)) {
+                    throw new ArgumentsRequired (this.id + " fetchTrades() requires 'lastId' id in params");
+                }
+                const response = await this[method] (this.deepExtend (request, params));
+                // {
+                //     success: true,
+                //     message: "",
+                //     result: [
+                //         {
+                //             id: 99601,
+                //             type: "buy",
+                //             time: 1640108002.455,
+                //             amount: "323144.05252658",
+                //             price: "0.008",
+                //             total: "2585.15242021"
+                //         },
+                //         {
+                //             id: 99544,
+                //             type: "buy",
+                //             time: 1640107990.112,
+                //             amount: "967.91335",
+                //             price: "0.00794371",
+                //             total: "7.68882295"
+                //         },
+                data = this.safeValue (response, 'result', []);
+            } else if (method === 'spotPublicGetHistoryResult') {
+                if (!('since' in params)) {
+                    throw new ArgumentsRequired (this.id + " fetchTrades() requires 'since' id in params");
+                }
+                data = this[method] (this.deepExtend (request, params));
+                // [
+                //     {
+                //         tid: 99601,
+                //         type: "buy",
+                //         date: 1640108002,
+                //         amount: "323144.05252658",
+                //         price: "0.008",
+                //         total: "2585.15242021"
+                //     },
+                //     {
+                //         tid: 99544,
+                //         type: "buy",
+                //         date: 1640107990,
+                //         amount: "967.91335",
+                //         price: "0.00794371",
+                //         total: "7.68882295"
+                //     },
+                if (!Array.isArray (data)) { // NOTE: Instead of this.safeValue I've used this
```

Comment:
We can remove this, no need to check the response type here.

Reply:
done. (actually i havent experienced, just I wanted to make the returned response fail-safe, because of uncertainty). however, removed.